autogen/graph_utils.py

```diff
@@ -0,0 +1,192 @@
+        if len(full_anti_join) > 0:
+            raise ValueError(
+                f"""allowed_speaker_transitions_dict has self-loops not mentioned in the list of agents allowed to repeat in allow_repeat_speaker_names. allow_repeat_speaker_names: {allow_repeat_speaker_names}; self_loop_agents: {self_loop_agents}
+                """
+            )
+
+    # Warnings
+    # Warning 1. Warning if there are isolated agent nodes, there are not incoming nor outgoing edges
+    # Concat keys if len(value) is positive
+    # Use get_successor_agent_names to get all agents that the agent can transition to
+    has_outgoing_edge = []
+    for key in allowed_speaker_transitions_dict.keys():
+        if len(get_successor_agent_names(key, allowed_speaker_transitions_dict)) > 0:
+            has_outgoing_edge.append(key)
+    no_outgoing_edges = [agent.name for agent in agents if agent.name not in has_outgoing_edge]
+
+    # allowed_speaker_transitions_dict.values() is a list of list of Agents
+    # values_all_agents is a list of all agents in allowed_speaker_transitions_dict.values()
+    # Use get_predecessor_agent_names to get all agents that can transition to the agent
+    has_incoming_edge = []
+    for agent_list in allowed_speaker_transitions_dict.values():
+        if len(agent_list) > 0:
+            has_incoming_edge.extend([agent.name for agent in agent_list])
+
+    no_incoming_edges = [agent.name for agent in agents if agent.name not in has_incoming_edge]
+
+    isolated_agents = set(no_incoming_edges).intersection(set(no_outgoing_edges))
+    if len(isolated_agents) > 0:
+        logging.warning(
+            f"""Warning: There are isolated agent nodes, there are not incoming nor outgoing edges. Isolated agents: {isolated_agents}"""
+        )
+
+    # Warning 2. Warning if the set of agents in allowed_speaker_transitions do not match agents
+    # Get set of agents
+    agents_in_allowed_speaker_transitions = set(has_incoming_edge).union(set(has_outgoing_edge))
+    agents_names = [agent.name for agent in agents]
+    full_anti_join = set(agents_in_allowed_speaker_transitions).symmetric_difference(set(agents_names))
+    if len(full_anti_join) > 0:
+        logging.warning(
+            f"""Warning: The set of agents in allowed_speaker_transitions do not match agents. Offending agents: {full_anti_join}"""
+        )
+
+
+def invert_disallowed_to_allowed(disallowed_speaker_transitions_dict: dict, agents: List[Agent]) -> dict:
+    """
+    Start with a fully connected allowed_speaker_transitions_dict of all agents. Remove edges from the fully connected allowed_speaker_transitions_dict according to the disallowed_speaker_transitions_dict to form the allowed_speaker_transitions_dict.
+    """
+    # Create a fully connected allowed_speaker_transitions_dict of all agents
+    allowed_speaker_transitions_dict = {agent.name: [other_agent for other_agent in agents] for agent in agents}
```

Comment:
```suggestion
    allowed_speaker_transitions_dict = {agent: agents for agent in agents}
```

Reply:
Is this related to "Consistency"? If yes, I will refactor `allowed_or_disallowed_speaker_transitions` and `allowed_speaker_transitions_dict` after addressing other issues.

Reply:
yes